optimum/intel/ipex/modeling_base.py

```diff
@@ -86,10 +87,37 @@ def _is_patched_with_ipex(model, task):
 
 
 def _prepare_inputs_for_ipex_model(model, task, use_cache):
-    if task in _IPEX_EXPORTED_GENERATION_TASKS and _is_patched_with_ipex(model, task):
-        return get_dummy_input(model, return_dict=True)
-    else:
-        return prepare_jit_inputs(model, task, use_cache)
+    sample_inputs = prepare_jit_inputs(model, task, use_cache)
+    if (
+        task in _IPEX_EXPORTED_GENERATION_TASKS
+        and _is_patched_with_ipex(model, task)
+        and "past_key_values" in sample_inputs
+    ):
+        # Only consider llama for now
+        assert len(sample_inputs["past_key_values"][0][0].shape) == 4
+        max_position = model.config.max_position_embeddings
+        batch_size = sample_inputs["input_ids"].shape[0]
+        past_length = sample_inputs["past_key_values"][0][0].shape[2]
+        num_attn = sample_inputs["past_key_values"][0][0].shape[1]
+        d_k = sample_inputs["past_key_values"][0][0].shape[-1]
+        dtype = sample_inputs["past_key_values"][0][0].dtype
+
+        num_layers = len(sample_inputs["past_key_values"])
+        beam_idx_tmp = torch.zeros((max_position, batch_size), dtype=torch.long).contiguous()
+        past_key_values = tuple(
+            [
+                (
+                    torch.zeros(1, past_length, past_length, 1, dtype=torch.long).contiguous(),
```

Comment:
what does this value correspond to ?

Reply:
It's meaningless, this is for keeping the original past length to match transformers functions like [_get_initial_cache_position](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/utils.py#L1401)